storage/remote/otlptranslator/prometheus/normalize_name.go

```diff
@@ -1,9 +1,20 @@
-// DO NOT EDIT. COPIED AS-IS. SEE ../README.md
-
-// Copyright The OpenTelemetry Authors
-// SPDX-License-Identifier: Apache-2.0
-
-package prometheus // import "github.com/prometheus/prometheus/storage/remote/otlptranslator/prometheus"
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// Provenance-includes-location: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/translator/prometheus/normalize_name.go
```

Comment:
I'm not super familiar with the Provenance-includes format.  Would it be good to reference a commit hash in-case the file name changes?

Reply:
We discussed whether to use a commit hash - we just went with the branch name, since that's what Mimir had done for its Cortex fork (we followed its example). We are waiting on some feedback from our licensing expertise, maybe they can clarify whether this matters :)

Reply:
I think a commit hash is preferable as it is immutable and, for provenance purposes especially, it is important to know exactly what you're referencing.  I highly doubt this code will ever be re-licensed, but if it were then a reference to a floating commit via branch name could become wildly wrong.

Reply:
Arve and I had this question while we were putting up the PR too. Thanks for taking a peek, I just added the hash references in the last commit :+1: